specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1178 @@
+data.Status(L"Downloaded!"); // Binds to {progressStatus} in xml payload
+
+auto result = co_await winrt::Microsoft::Windows::AppNotifications::AppNotificationManager::Default().UpdateAsync(data, L"Tag", L"Group");
+```
+
+![Progress Bar Example 2](ProgressBarExample2.png)
+
+# Retrieving Arguments
+
+AppNotificationContent and Button return arguments to the activated application when the user clicks
+on the component. Developers can add multiple key/value pairs and Windows App SDK will serialize
+them into a string.
+
+Below is an example with AppNotificationContent and Button:
+
+```cpp
+AppNotificationContent()
+    .AddArgument(L"action", L"appNotificationBodyTouch")
+    .AddArgument(L"user", 938163)
+    .AddButton(AppNotificationButton(L"Send")
+        .AddArgument(L"action", L"buttonTouch")
+        .AddArgument(L"user", 938163));
+```
+
+When the user clicks on the AppNotification body, the resulting argument string is:
+
+-   "action=appNotificationBodyTouch;user=938163"
+
+When the user clicks on the Button, the result argument string is:
+
+-   "action=buttonTouch;callId=938163"
+
+Notice that ';' is used as the delimiter between each argument.
+
+To retrieve the arguments from the serialized string, developers will use
+AppNotificationReceivedEventArgs::Arguments to get the values from the argument string as a map.
+
+Below is an example usage:
+
+```cpp
+using namespace winrt::Microsoft::Windows::AppLifecycle;
+using namespace winrt::Microsoft::Windows::AppNotifications;
+
+// Retrieve the activation args using Windows App SDK AppLifecycle functions
+AppActivationArguments args { AppLifecycle::AppInstance::GetCurrent().GetActivatedEventArgs() };
+ExtendedActivationKind kind { args.Kind() };
+
+// Check the activation kind
+if (kind == ExtendedActivationKind::AppNotification)
+{
+    AppNotificationActivatedEventArgs appActivatedArgs { args.Data().as<AppNotificationActivatedEventArgs>() };
+
+    // Retrieve the serialized arguments from AppNotificationActivatedEventArgs
+    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring> argumentMap { appActivatedArgs.Argument() };
+}
+```
+
+# Full API Details
+
+```cpp
+namespace Microsoft.Windows.AppNotifications.Builder
+{
+    runtimeclass AppNotificationTextProperties
+    {
+        // Contains the set of <text> attributes
+        AppNotificationTextProperties();
+
+        AppNotificationTextProperties SetLanguage(String value);
+        AppNotificationTextProperties UseIncomingCallAlignment();
+        AppNotificationTextProperties SetMaxLines(Int32 value);
+
+        String GetXml();
+    }
+
+    enum AppNotificationButtonStyle
+    {
+        Success,
+        Critical,
+    };
+
+    runtimeclass AppNotificationButton
+    {
+        // Button can use ToolTip instead of content which requires empty content.
+        // Argument attribute is required and uses the ArgumentSerializer construct
```

Comment:
What is this "ArgumentSerializer"? Is that covered by some other spec?

Reply:
Was from old iteration of spec. Removing.